src/applications/coronavirus-vaccination/components/Unsubscribe.jsx

```diff
@@ -0,0 +1,72 @@
+import React, { useEffect, useState } from 'react';
+import { connect } from 'react-redux';
+import { withRouter } from 'react-router';
+import { requestStates } from 'platform/utilities/constants';
+import LoadingIndicator from '@department-of-veterans-affairs/formation-react/LoadingIndicator';
+
+import { unsubscribe } from '../api';
+
+import { focusElement } from 'platform/utilities/ui';
+
+// Assumes URL is like /unsubscribe?sid={sid}
+function Unsubscribe({ router }) {
+  const sid = { sid: router.location.query.sid };
+
+  const [unsubscribeStatus, setUnsubscribeStatus] = useState(
+    requestStates.notCalled,
+  );
+
+  useEffect(() => {
+    focusElement('#covid-vaccination-heading-unsubscribe');
+    async function unsubscribeBySid() {
+      try {
+        await unsubscribe(sid);
+        setUnsubscribeStatus(requestStates.succeeded);
+      } catch (error) {
+        setUnsubscribeStatus(requestStates.failed);
+      }
+    }
+    unsubscribeBySid();
+  }, []);
```

Comment:
`sid` shouldn't ever change so it doesn't _really_ matter but I think in the dependencies array here you should have the `sid` listed 

```js
  useEffect(() => {
    focusElement('#covid-vaccination-heading-unsubscribe');
    async function unsubscribeBySid() {
      try {
        await unsubscribe(sid);
        setUnsubscribeStatus(requestStates.succeeded);
      } catch (error) {
        setUnsubscribeStatus(requestStates.failed);
      }
    }
   
    if (sid) {
        unsubscribeBySid();
    } else {
        // I think "router.location.query.sid " should be defined on first execution so this if-statement shouldn't
        // really matter, but it could happen if there is no SID in the search bar.
    }
  }, [sid]);
```

You could also go a step further and abstract the whole thing into a hook.

```js 
// hooks/unsubscribe.js

export function unsubscribe(sid) {
  const [unsubscribeStatus, setUnsubscribeStatus] = useState(
    requestStates.notCalled,
  );

  useEffect(() => {
   // focusElement could be called in a separate useEffect 
    async function unsubscribeBySid() {
      try {
        await unsubscribe(sid);
        setUnsubscribeStatus(requestStates.succeeded);
      } catch (error) {
        setUnsubscribeStatus(requestStates.failed);
      }
    }
    unsubscribeBySid();
  }, [sid]);
 
  return [unsubscribeStatus]
}

```